### DISCUSSION TITLE
Should console.log work with the new javascript module?

### DISCUSSION THREAD

**Discussion by well-thats-funny** [stashapp/stash#4936] [Q&A] [answered]
I'm currently working on some xPathScrapers using Javascript for some postprocessing. After updating to Stash 0.26, I can't use console.log anymore:

`WARN[2024-06-06 10:26:30] javascript failed to run: ReferenceError: console is not defined at <eval>:2:3(6)`

I believe it's due to a change of the JS engine (https://github.com/stashapp/stash/pull/4631), because it worked fine with 0.25.1 (it just printed to stdout). Aren't we expected to use console.log in the postprocessing scripts? It's a quite useful feature for debugging.

**Comment by Maista6969** [comment]
I looked at the [new Javascript package](https://github.com/stashapp/stash/tree/develop/pkg/javascript) and it appears to have a module for logging but it is only used for Javascript plugins. I have opened #4937 to add this to scrapers as well 👍

**Comment by WithoutPants** [comment, marked as answer]
This is a regression on existing behaviour, and I'll address it in a bugfix release. #4937 will go into 0.27.
